### PR TITLE
Consider the key's lowercase form while grouping

### LINKED ISSUE #85: utils.groupBy should be case-insensitive
On some websites, the categories on which we're grouping on don't share the same case. For the grouping, it makes a lot of sense to actually consider the lowercase form only.

### PR DESCRIPTION
**Pull request by redox**
Fix #85

### CODE REVIEW

**Comment by redox** [comment]
@pixelastic What do you think about that?

File: src/lib/utils.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ let utils = {
       if (item[property] === undefined) {
         throw new Error(`[groupBy]: Object has no key ${property}`);
       }
-      let key = item[property];
+      let key = item[property] && item[property].toLowerCase();
```
Comment (pixelastic) [inline src/lib/utils.js]: Why the `&&`? 

`let key = item[property].toLowerCase();` should be enough

Comment (maxiloc) [inline src/lib/utils.js]: I think playground was failing otherwise

Comment (redox) [inline src/lib/utils.js]: Because `item[property]` can be `null`

Comment (pixelastic) [inline src/lib/utils.js]: Then we should update the `if` a few line above, where it is testing if it's `undefined` to also test `null`

Comment (redox) [inline src/lib/utils.js]: Hmmm, I think it's fine to group hits sharing the `null` value together. It happens if - for a unknown reason - you don't have `lvl1` for instance.

I can make the test more explicit if you want.

Comment (pixelastic) [inline src/lib/utils.js]: I would make the code more explicit then.

```
let key = item[property];
if (key.toLowerCase) {
  let = key.toLowerCase();
}
```

Still not perfect (`key` can be an object with a `toLowerCase` field that is not a function...) but it expresses the intent of what you want to do more easily.
